File: Flutter(Frontend)/mk_bank_project/lib/entity/fixed_deposit_model.dart
class FixedDeposit {
  int? id;
  double? depositAmount;
  int? durationInMonths;
  double? interestRate;
  double? prematureInterestRate;
  String? startDate;
  String? maturityDate;
  double? maturityAmount;
  String? prematureWithdrawalDate;
  String? status;
  String? lastUpdatedAt;
  AccountInfo? account;

  FixedDeposit({
    this.id,
    this.depositAmount,
    this.durationInMonths,
    this.interestRate,
    this.prematureInterestRate,
    this.startDate,
    this.maturityDate,
    this.maturityAmount,
    this.prematureWithdrawalDate,
    this.status,
    this.lastUpdatedAt,
    this.account,
  });

  factory FixedDeposit.fromJson(Map<String, dynamic> json) {
    return FixedDeposit(
      id: json['id'],
      depositAmount: (json['depositAmount'] as num?)?.toDouble(),
      durationInMonths: json['durationInMonths'],
      interestRate: (json['interestRate'] as num?)?.toDouble(),
      prematureInterestRate: (json['prematureInterestRate'] as num?)?.toDouble(),
      startDate: json['startDate'],
      maturityDate: json['maturityDate'],
      maturityAmount: (json['maturityAmount'] as num?)?.toDouble(),
      prematureWithdrawalDate: json['prematureWithdrawalDate'],
      status: json['status'],
      lastUpdatedAt: json['lastUpdatedAt'],
      account: json['account'] != null ? AccountInfo.fromJson(json['account']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'depositAmount': depositAmount,
      'durationInMonths': durationInMonths,
    };
  }
}

class AccountInfo {
  int id;
  String name;
  String accountType;
  double balance;

  AccountInfo({
    required this.id,
    required this.name,
    required this.accountType,
    required this.balance,
  });

  factory AccountInfo.fromJson(Map<String, dynamic> json) {
    return AccountInfo(
      id: json['id'],
      name: json['name'],
      accountType: json['accountType'],
      balance: (json['balance'] as num).toDouble(),
    );
  }
}

File: Angular(Frontend)/mkbank/src/app/model/fixedDeposit.model.ts
import { Accounts } from "./accounts.model";
import { FdStatus } from "./fdStatus";

export class FixedDeposit {

  // id?: number;
  // // account!: Accounts;
  // account?: any;
  // depositAmount!: number;
  // durationInMonths!: number;
  // interestRate!: number;
  // prematureInterestRate!: number;
  // startDate!: Date;
  // maturityDate!: Date;
  // maturityAmount!: number;
  // prematureWithdrawalDate?: Date | null;
  // status!: FdStatus;
  // lastUpdatedAt?: string;
  // fDLustUpdatedAt!: Date;



  id!: number;
  depositAmount?: number;
  durationInMonths?: number;
  interestRate?: number;
  prematureInterestRate?: number;
  startDate?: string;
  maturityDate?: string;
  maturityAmount?: number;
  prematureWithdrawalDate?: string | null;
  status?: string;
  lastUpdatedAt?: string | null;
  account?: {
    id: number;
    name: string;
    accountType: string;
    balance: number;
  };
}

